File: ox_push/ios/Classes/OXPushPlugin.m
#import "OXPushPlugin.h"

@implementation OXPushPlugin

static FlutterMethodChannel* _channel = nil;

+ (void)registerWithRegistrar:(NSObject<FlutterPluginRegistrar>*)registrar {
  _channel = [FlutterMethodChannel
      methodChannelWithName:@"ox_push"
            binaryMessenger:[registrar messenger]];
  OXPushPlugin* instance = [[OXPushPlugin alloc] init];
  [registrar addMethodCallDelegate:instance channel:_channel];
}

- (void)handleMethodCall:(FlutterMethodCall*)call result:(FlutterResult)result {
  if ([@"getPlatformVersion" isEqualToString:call.method]) {
    result([@"iOS " stringByAppendingString:[[UIDevice currentDevice] systemVersion]]);
  } else {
    result(FlutterMethodNotImplemented);
  }
}

+ (FlutterMethodChannel*)channel {
  return _channel;
}

@end

File: packages/base_framework/ox_push/ios/Classes/OXPushPlugin.swift
import Flutter
import UIKit

public class OXPushPlugin: NSObject, FlutterPlugin {
    
    public static var channel: FlutterMethodChannel?
    
    public static func register(with registrar: FlutterPluginRegistrar) {
        let channel = FlutterMethodChannel(name: "ox_push", binaryMessenger: registrar.messenger())
        let instance = OXPushPlugin()
        registrar.addMethodCallDelegate(instance, channel: channel)
        self.channel = channel
    }
    
    public func handle(_ call: FlutterMethodCall, result: @escaping FlutterResult) {
        switch call.method {
        case "getPlatformVersion":
            result("iOS " + UIDevice.current.systemVersion)
        default:
            result(FlutterMethodNotImplemented)
        }
    }
}
